stacks_queues/queue.py

```diff
@@ -15,47 +15,80 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
+
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException("Queue is full")
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
```

Comment:
👍

If the size returns to 0, consider resetting front and rear (setting them back to the initialized state).